### DISCUSSION TITLE
Livewire loses route-bound model

### DISCUSSION THREAD

**Discussion by mityukov** [livewire/livewire#2996] [7. Bugs (archived)]
### Description

After clicking the "Adjust" button, I get "Call to a member function resolveRouteBinding() on null" error.

Note: when originally rendering the component — it looks that route-bound model is Ok.

Here's the important code parts:
![image](https://user-images.githubusercontent.com/1751910/121348075-f3c41e80-c938-11eb-899d-8c7331100b2e.png)

Can't figure out what's wrong exactly…

**Context**
- Livewire version: `versions : * v2.4.4` (not sure what that asterisk means exactly)
- Laravel version: 8.44.0
- Alpine version: 2.7.3
- Browser: Chrome & Firefox

**Comment by mityukov** [comment]
The request payload:
```json
{"fingerprint":{"id":"pbqUS9JRDEuH4Y2oSlgm","name":"admin.adjust-balance","locale":"ru","path":"admin/billing/3","method":"GET"},"serverMemo":{"children":[],"errors":[],"htmlHash":"aa710f63","data":{"account":[],"showModal":false},"dataMeta":{"models":{"account":{"class":"App\\Account","id":3,"relations":["plan"],"connection":"mysql"}}},"checksum":"b87d6abbaa86b8be910b29f9f902db6b89f4928487596dcb744a1cc9775d0d0a"},"updates":[{"type":"callMethod","payload":{"method":"openAdjustModal","params":[]}}]}
```

**Comment by mityukov** [comment]
Ok, it looks it's not "route-bound"; it's a model, passed into the component via blade:
```blade
<livewire:admin.adjust-balance :account="$account" />
```

Still, how do I make it working?…

**Comment by mityukov** [comment]
I've removed route model building from the Page itself (the page, where this livewire component is called) — and Livewire is working =)

was:
```php
class BillingController extends Controller
{
    public function index(Account $account)
    {
        return view('super.accounts.billing', [
            'account' => $account,
            'plans' => Plan::where('active', true)->get(),
        ]);
    }
```


now:
```php
class BillingController extends Controller
{
    public function index($account) // now account is an Id
    {
        return view('super.accounts.billing', [
            'account' => Account::find($account),
            'plans' => Plan::where('active', true)->get(),
        ]);
    }
```

So, it failed on Laravel routing end, but why it even tried to "route-bind" anything on a POST request to `/livewire/message/admin.adjust-balance` URL??

**Comment by dakira** [comment]
@mityukov Don't you think this should go into [Discussions](https://github.com/livewire/livewire/discussions)?

**Comment by mityukov** [comment]
@dakira actually, still not sure it's not a bug.

> ... but why it even tried to "route-bind" anything on a POST request to /livewire/message/admin.adjust-balance URL??

To my understanding, it's a Livewire widget I'm working with right now (when `wire:click` fires). Why does it fail on something, related to the controller of the page it is contained within?
